python/aitemplate/backend/builder.py

```diff
@@ -844,14 +844,16 @@ def _gen_compiler_version_files(self, target_dir):
         # for cache invalidation purposes (different compiler versions
         # should not reuse same cached build artifacts )
         cc = Target.current().cc()
-        compilers = {"main_compiler": cc}
+        compilers = {}
         if "nvcc" in cc:
-            ccbin_match = re.search(r'-ccbin "?([^ "]+)', cc)
+            ccbin_match = re.search(r'(.*) -ccbin "?([^ "]+)', cc)
             if ccbin_match:
-                nvcc_host_compiler = ccbin_match.group(1)
+                cc = ccbin_match.group(1)
+                nvcc_host_compiler = ccbin_match.group(2)
             else:
                 nvcc_host_compiler = "g++"  # default, using PATH resolution
             compilers["nvcc_host_compiler"] = nvcc_host_compiler
+        compilers["main_compiler"] = cc
```

Comment:
Previously, when `cc` contained `-ccbin ...` in it, the latter was preserved in the `compilers["main_compiler"]` value. With the updated logic, the part starting from `-ccbin` will be cut: i.e., the `compilers["main_compiler"]` will be just `"nvcc"`.

I'm wondering if this was a bug in the original code and now it's fixed, because this PR seems to change the previous behavior.

Reply:
Yes, it was a bug.
The subsequent code will check (both host and nvcc) compiler version by `cc --version` (https://github.com/facebookincubator/AITemplate/blob/main/python/aitemplate/backend/builder.py#L860)
and it would produce error if ccbin is include in cc (previous: `nvcc -ccbin g++ --version`, but it should be `nvcc --version`)

Reply:
But, the bug was not triggered previously since no code path set ccbin in cc. I thought this code segment might be a copy-paste from fb's internal source or someone added during manual debugging.